### DISCUSSION TITLE
Version 0.2.0 release

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [somnathrakshit/geograpy3#57] [Show and tell]
Today version 0.2.0 of geograpy3 is released which is a major improvement and change.
All lookup is now done against country,region and city datasets derived from wikidata.

**Comment by WolfgangFahl** [comment]
**LabelLookup example #1**
**query**
```sql
select  distinct wikidataid,label,name,pop,regionIso,regionName,countryIso,countryName,gndId,geoNameId,lat,lon from
CityLookup cl
where label in ('Berlin',',St. Petersburg','Singapore','Athens')
order by pop desc

```
**result**
| wikidataid                                           | label     | name             |     pop | regionIso   | regionName          | countryIso   | countryName              | gndId     |   geoNameId |   lat |   lon |
|------------------------------------------------------|-----------|------------------|---------|-------------|---------------------|--------------|--------------------------|-----------|-------------|-------|-------|
| [Q334](https://www.wikidata.org/wiki/Q334)           | Singapore | Singapore        | 5888926 | SG          | Singapore           | SG           | Singapore                | 4055089-8 |     1880251 |   104 |     1 |
| [Q64](https://www.wikidata.org/wiki/Q64)             | Berlin    | Berlin           | 3644826 | DE-BE       | Berlin              | DE           | Germany                  | 4005728-8 |     6547383 |    53 |    13 |
| [Q64](https://www.wikidata.org/wiki/Q64)             | Berlin    | Berlin           | 3644826 | DE-BE       | Berlin              | DE           | Germany                  | 4005728-8 |     6547539 |    53 |    13 |
| [Q64](https://www.wikidata.org/wiki/Q64)             | Berlin    | Berlin           | 3644826 | DE-BE       | Berlin              | DE           | Germany                  | 4005728-8 |     2950159 |    53 |    13 |
| [Q64](https://www.wikidata.org/wiki/Q64)             | Berlin    | Berlin           | 3644826 | DE-BE       | Berlin              | DE           | Germany                  | 4005728-8 |     2950157 |    53 |    13 |
| [Q1524](https://www.wikidata.org/wiki/Q1524)         | Athens    | Athens           |  664046 | GR-I        | Attica Region       | GR           | Greece                   | 4003366-1 |      264371 |    38 |    24 |
| [Q203263](https://www.wikidata.org/wiki/Q203263)     | Athens    | Athens           |  115452 | US-GA       | Georgia             | US           | United States of America | 4195479-8 |     4180386 |    34 |   -83 |
| [Q79439](https://www.wikidata.org/wiki/Q79439)       | Athens    | Athens           |   24000 | US-AL       | Alabama             | US           | United States of America |           |     4830668 |    35 |   -87 |
| [Q755420](https://www.wikidata.org/wiki/Q755420)     | Athens    | Athens           |   23832 | US-OH       | Ohio                | US           | United States of America | 4207790-4 |     4505542 |    39 |   -82 |
| [Q755420](https://www.wikidata.org/wiki/Q755420)     | Athens    | Athens           |   21342 | US-OH       | Ohio                | US           | United States of America | 4207790-4 |     4505542 |    39 |   -82 |
| [Q821199](https://www.wikidata.org/wiki/Q821199)     | Berlin    | Berlin           |   19866 | US-CT       | Connecticut         | US           | United States of America |           |     5282244 |    42 |   -73 |
| [Q3292481](https://www.wikidata.org/wiki/Q3292481)   | Athens    | Athens           |   13548 | US-TN       | Tennessee           | US           | United States of America |           |     4603284 |    35 |   -85 |
| [Q3292481](https://www.wikidata.org/wiki/Q3292481)   | Athens    | Athens           |   13458 | US-TN       | Tennessee           | US           | United States of America |           |     4603284 |    35 |   -85 |
| [Q755425](https://www.wikidata.org/wiki/Q755425)     | Athens    | Athens           |   12710 | US-TX       | Texas               | US           | United States of America |           |     4671545 |    32 |   -96 |
| [Q821244](https://www.wikidata.org/wiki/Q821244)     | Berlin    | Berlin           |   10051 | US-NH       | New Hampshire       | US           | United States of America |           |     5083330 |    44 |   -71 |
| [Q1086827](https://www.wikidata.org/wiki/Q1086827)   | Berlin    | Berlin           |    7588 | US-NJ       | New Jersey          | US           | United States of America |           |     4500771 |    40 |   -75 |
| [Q1569850](https://www.wikidata.org/wiki/Q1569850)   | Berlin    | Berlin           |    5524 | US-WI       | Wisconsin           | US           | United States of America |           |     5245497 |    44 |   -89 |
| [Q3720748](https://www.wikidata.org/wiki/Q3720748)   | Berlin    | Berlin Township  |    5357 | US-NJ       | New Jersey          | US           | United States of America |           |     4500777 |    40 |   -75 |
| [Q614184](https://www.wikidata.org/wiki/Q614184)     | Berlin    | Berlin           |    4485 | US-MD       | Maryland            | US           | United States of America |           |     4348460 |    38 |   -75 |
| [Q3709146](https://www.wikidata.org/wiki/Q3709146)   | Athens    | Athens           |    4089 | US-NY       | New York            | US           | United States of America |           |     5107467 |    42 |   -74 |
| [Q570807](https://www.wikidata.org/wiki/Q570807)     | Athens    | Athens           |    3013 | CA-ON       | Ontario             | CA           | Canada                   |           |     5887943 |    45 |   -76 |
| [Q524646](https://www.wikidata.org/wiki/Q524646)     | Berlin    | Berlin           |    2866 | US-MA       | Massachusetts       | US           | United States of America |           |     4930431 |    42 |   -72 |
| [Q578681](https://www.wikidata.org/wiki/Q578681)     | Athens    | Athens           |    1988 | US-IL       | Illinois            | US           | United States of America |           |     4232997 |    40 |   -90 |
| [Q821215](https://www.wikidata.org/wiki/Q821215)     | Berlin    | Berlin           |    1880 | US-NY       | New York            | US           | United States of America |           |     5108863 |    43 |   -73 |
| [Q4892348](https://www.wikidata.org/wiki/Q4892348)   | Berlin    | Berlin           |    1145 | US-WI       | Wisconsin           | US           | United States of America |           |             |    44 |   -89 |
| [Q977715](https://www.wikidata.org/wiki/Q977715)     | Athens    | Athens           |    1105 | US-WI       | Wisconsin           | US           | United States of America |           |     5244312 |    45 |   -90 |
| [Q2218754](https://www.wikidata.org/wiki/Q2218754)   | Athens    | Athens           |    1048 | US-WV       | West Virginia       | US           | United States of America |           |     4797549 |    37 |   -81 |
| [Q2218754](https://www.wikidata.org/wiki/Q2218754)   | Athens    | Athens           |    1048 | US-WV       | West Virginia       | US           | United States of America |           |     4797549 |    37 |   -81 |
| [Q2791349](https://www.wikidata.org/wiki/Q2791349)   | Athens    | Athens           |    1024 | US-MI       | Michigan            | US           | United States of America |           |     4984489 |    42 |   -85 |
| [Q3477164](https://www.wikidata.org/wiki/Q3477164)   | Athens    | Athens           |    1019 | US-ME       | Maine               | US           | United States of America |           |     4956946 |    45 |   -70 |
| [Q4892350](https://www.wikidata.org/wiki/Q4892350)   | Berlin    | Berlin           |     945 | US-WI       | Wisconsin           | US           | United States of America |           |     5245512 |    45 |   -90 |
| [Q142659](https://www.wikidata.org/wiki/Q142659)     | Berlin    | Berlin           |     898 | US-OH       | Ohio                | US           | United States of America |           |     5147132 |    41 |   -82 |
| [Q2345711](https://www.wikidata.org/wiki/Q2345711)   | Berlin    | Berlin           |     551 | US-GA       | Georgia             | US           | United States of America |           |     4182096 |    31 |   -84 |
| [Q3239684](https://www.wikidata.org/wiki/Q3239684)   | Athens    | Athens           |     249 | US-LA       | Louisiana           | US           | United States of America |           |     4315137 |    33 |   -93 |
| [Q1930098](https://www.wikidata.org/wiki/Q1930098)   | Berlin    | Lincoln          |     162 | US-IA       | Iowa                | US           | United States of America |           |     4864742 |    42 |   -93 |
| [Q821229](https://www.wikidata.org/wiki/Q821229)     | Berlin    | Berlin           |      34 | US-ND       | North Dakota        | US           | United States of America |           |     5058265 |    46 |   -98 |
| [Q12303254](https://www.wikidata.org/wiki/Q12303254) | Berlin    | Berlin           |         | DK-83       | Southern Denmark    | DK           | Denmark                  |           |             |    55 |    10 |
| [Q844930](https://www.wikidata.org/wiki/Q844930)     | Athens    | Classical Athens |         | GR-I        | Attica Region       | GR           | Greece                   |           |     9962195 |    38 |    24 |
| [Q20144232](https://www.wikidata.org/wiki/Q20144232) | Berlin    | Berlín           |         | MX-CHP      | Chiapas             | MX           | Mexico                   |           |     8922112 |    16 |   -93 |
| [Q582242](https://www.wikidata.org/wiki/Q582242)     | Berlin    | Berlín           |         | SV-US       | Usulután Department | SV           | El Salvador              |           |     3587266 |    14 |   -89 |
| [Q4892339](https://www.wikidata.org/wiki/Q4892339)   | Berlin    | Berlin           |         | US-AL       | Alabama             | US           | United States of America |           |     4047787 |    34 |   -87 |
| [Q4813438](https://www.wikidata.org/wiki/Q4813438)   | Athens    | Athens           |         | US-AR       | Arkansas            | US           | United States of America |           |     4099912 |    34 |   -94 |
| [Q4813442](https://www.wikidata.org/wiki/Q4813442)   | Athens    | Athens           |         | US-CA       | California          | US           | United States of America |           |     5325144 |    34 |  -118 |
| [Q2504681](https://www.wikidata.org/wiki/Q2504681)   | Berlin    | Berlin           |         | US-IL       | Illinois            | US           | United States of America |           |     4233627 |    40 |   -90 |
| [Q4813446](https://www.wikidata.org/wiki/Q4813446)   | Athens    | Athens           |         | US-IN       | Indiana             | US           | United States of America |           |     4917703 |    41 |   -86 |
| [Q17509488](https://www.wikidata.org/wiki/Q17509488) | Berlin    | Berlin           |         | US-KS       | Kansas              | US           | United States of America |           |             |    38 |   -95 |
| [Q42685464](https://www.wikidata.org/wiki/Q42685464) | Athens    | Athens           |         | US-KS       | Kansas              | US           | United States of America |           |             |    38 |   -96 |
| [Q821190](https://www.wikidata.org/wiki/Q821190)     | Berlin    | Berlin           |         | US-KY       | Kentucky            | US           | United States of America |           |     4283975 |    39 |   -84 |
| [Q4813443](https://www.wikidata.org/wiki/Q4813443)   | Athens    | Athens           |         | US-KY       | Kentucky            | US           | United States of America |           |     4282781 |    38 |   -84 |
| [Q33422655](https://www.wikidata.org/wiki/Q33422655) | Athens    | Athens           |         | US-ME       | Maine               | US           | United States of America |           |     4956946 |    45 |   -70 |
| [Q7522845](https://www.wikidata.org/wiki/Q7522845)   | Singapore | Singapore        |         | US-MI       | Michigan            | US           | United States of America |           |             |    43 |   -86 |
| [Q4813447](https://www.wikidata.org/wiki/Q4813447)   | Athens    | Athens           |         | US-MN       | Minnesota           | US           | United States of America |           |     5016799 |    45 |   -93 |
| [Q4813449](https://www.wikidata.org/wiki/Q4813449)   | Athens    | Athens           |         | US-MS       | Mississippi         | US           | United States of America |           |     4416664 |    34 |   -88 |
| [Q4892354](https://www.wikidata.org/wiki/Q4892354)   | Berlin    | Berlin           |         | US-OK       | Oklahoma            | US           | United States of America |           |     4530350 |    35 |  -100 |
| [Q1130437](https://www.wikidata.org/wiki/Q1130437)   | Athens    | Athens           |         | US-PA       | Pennsylvania        | US           | United States of America |           |     5178651 |    42 |   -77 |
| [Q1130437](https://www.wikidata.org/wiki/Q1130437)   | Athens    | Athens           |         | US-PA       | Pennsylvania        | US           | United States of America |           |     5178651 |    42 |   -77 |
| [Q1130950](https://www.wikidata.org/wiki/Q1130950)   | Berlin    | Berlin           |         | US-PA       | Pennsylvania        | US           | United States of America |           |     4556518 |    40 |   -79 |
| [Q1130950](https://www.wikidata.org/wiki/Q1130950)   | Berlin    | Berlin           |         | US-PA       | Pennsylvania        | US           | United States of America |           |     4556518 |    40 |   -79 |
| [Q21196360](https://www.wikidata.org/wiki/Q21196360) | Berlin    | Berlin           |         | US-TX       | Texas               | US           | United States of America |           |             |    30 |   -96 |
| [Q14711981](https://www.wikidata.org/wiki/Q14711981) | Athens    | Athens           |         | US-VA       | Virginia            | US           | United States of America |           |     4829249 |    38 |   -77 |
| [Q755427](https://www.wikidata.org/wiki/Q755427)     | Athens    | Athens           |         | US-VT       | Vermont             | US           | United States of America |           |     5233317 |    43 |   -73 |
| [Q821245](https://www.wikidata.org/wiki/Q821245)     | Berlin    | Berlin           |         | US-VT       | Vermont             | US           | United States of America |           |             |    44 |   -73 |
| [Q664745](https://www.wikidata.org/wiki/Q664745)     | Berlin    | Berlin           |         | US-WV       | West Virginia       | US           | United States of America |           |     4798708 |    39 |   -80 |
| [Q4892343](https://www.wikidata.org/wiki/Q4892343)   | Berlin    | Berlin           |         | ZA-EC       | Eastern Cape        | ZA           | South Africa             |           |             |   -33 |    28 |

**Comment by somnathrakshit** [comment]
https://github.com/somnathrakshit/geograpy3/blob/fcea745d89030496d51ef0234c1d7534e37cee82/geograpy/locator.py#L207

I think downloading from raw.githubusercontent.com is often unreliable and is leading to some of the bug reports that we have seen recently. However, downloading an asset from Github Releases is safer as seen here
https://github.com/somnathrakshit/geograpy3/blob/2c5fd0ef6b843e83099fb3d1fb47387652df1546/geograpy/locator.py#L1186

**Reply by somnathrakshit** [reply]
https://github.com/somnathrakshit/geograpy3/issues/59#issuecomment-923947897 maybe related to this.

**Comment by WolfgangFahl** [comment]
I've asked @tholzheim  to fix this using the  Download class create for our event corpus

**Reply by somnathrakshit** [reply]
Thank you!
